### ISSUE TITLE
macOS Sierra (Dev Beta 1) – Compatibility Issues

### ISSUE THREAD

**Issue by sifex** [rsms/fb-mac-messenger#302]
Hi All
Just checking in for those who are currently running the Beta and are having troubles with getting messenger to launch.

Currently, mine is stuck at 0% on the loading screen.

**Comment by rsms** [comment]
Interesting. This sounds similar to the performance issues some people have been experiencing in 10.11. 

Try this and then relaunch the app:

```
$ defaults write me.rsms.fbmessenger main.js/disable -bool YES
```

It disables loading/injecting of javascript. You'll lose most of the native functionality and the UI will look less than ideal, but if that works it might give us clues to what is going on.

To reset:

```
$ defaults delete me.rsms.fbmessenger main.js/disable
```

**Comment by emilstahl** [comment]
@rsms It doesn't work. 

```
Release 19:10:13.119381 Messenger   Created DB, header sequence number = 4264
Release 19:10:13.126815 Messenger   Created DB, header sequence number = 4264
Release 19:10:13.570424 Messenger   Reloading app
Release 19:10:14.576117 Messenger   Faulting in NSHTTPCookieStorage singleton
Release 19:10:14.576172 Messenger   Faulting in CFHTTPCookieStorage singleton
Release 19:10:14.576183 Messenger   Creating default cookie storage with process/bundle identifier
Release 19:10:14.585286 Messenger   TIC TCP Conn Start [1:0x6000001956e0]
Release 19:10:14.653309 Messenger   TIC TCP Conn Event [1:0x6000001956e0]: 1 Err(0)
Release 19:10:14.653341 Messenger   TIC TCP Conn Connected [1:0x6000001956e0]: Err(0)
Release 19:10:14.653483 Messenger   TIC Enabling TLS [1:0x6000001956e0]
Release 19:10:14.757713 Messenger   TIC TLS Event [1:0x6000001956e0]: 2, Pending(0)
Release 19:10:14.758262 Messenger   TIC TLS Event [1:0x6000001956e0]: 11, Pending(0)
Release 19:10:14.759842 Messenger   TIC TLS Event [1:0x6000001956e0]: 12, Pending(0)
Release 19:10:14.759896 Messenger   TIC TLS Event [1:0x6000001956e0]: 14, Pending(0)
Release 19:10:14.766177 Messenger   TIC TLS Trust Result [1:0x6000001956e0]: 0
Release 19:10:15.126761 Messenger   TIC TLS Event [1:0x6000001956e0]: 20, Pending(0)
Release 19:10:15.127122 Messenger   TIC TCP Conn Event [1:0x6000001956e0]: 8 Err(0)
Release 19:10:15.127145 Messenger   TIC TLS Handshake Complete [1:0x6000001956e0]
```

**Comment by sifex** [comment]
Found the fix. https://github.com/sifex/fb-mac-messenger/commit/479f1a38faa6fb8f50cc933579582c77f6d57a8d

Don't ask me why this fixed it. 

Edit: I'm not going to pull request this because this isn't good coding practice.

**Comment by rsms** [comment]
@sifex I saw you withdrew your PR. Looking at that line of change, I truly doubt it has anything to do with the 10.12 issues.

I just installed 10.12 on a spare macbook yesterday and are about to investigate what's going on now.

**Comment by emilstahl** [comment]
@rsms Believe it or not.. It fixed the issue. :)

**Comment by rsms** [comment]
The fix by @sifex indeed does fix the issue (at least partially.)
It appears that WebKit behavior changed in 10.12 where in <=10.11 the following would be true for a new WebView that had yet to be loaded:
`_webView.mainFrame.DOMDocument.URL.length == 0` though in 10.12, the initial value is "about:blank" which causes the app to reload "about:blank" instead of loading messenger.com

Further, this is strange... Adding the following to main.m:

``` m
#import <Cocoa/Cocoa.h>

int main(int argc, const char * argv[]) {
  printf("printf 1\n");
  @autoreleasepool {
    NSLog(@"NSLog 1");
    printf("printf 2\n");
    NSLog(@"NSLog 2");
  }
  return NSApplicationMain(argc, argv);
}
```

Causes the following output: (NSLog never writes to stdout or stderr)

```
printf 1
printf 2
```

Something really fishy is going on. I guess this is why 10.12 is clearly marked as pre-release beta :)

**Comment by rsms** [comment]
This has now been tested and released with version 0.1.8.1466363428
